core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -556,7 +569,7 @@ private void registerBuiltinQueryParsers() {
         registerQueryParser(new CommonTermsQueryParser(), CommonTermsQueryParser.QUERY_NAME_FIELD);
         registerQueryParser(new SpanMultiTermQueryParser(), SpanMultiTermQueryParser.QUERY_NAME_FIELD);
         QueryParser<FunctionScoreQueryBuilder> functionScoreParser = (QueryParseContext c) -> FunctionScoreQueryBuilder
-                .fromXContent((String name) -> functionScoreParsers.get(name), c);
+                .fromXContent(this::lookupScoreFunctionParser, c);
```

Comment:
I'm ok with making a registry but I'm going to say my peace first.

I *like* taking functional interfaces as parameters from a readability standpoint because it says "This is a thing I need to do but I can't do. If you want to call me you need to figure out how to do that." Taking a registry just says "I need a registry". I have to go look at the registry to figure out *why* it needs it.

I like the functional interfaces from a "minimal cohesion" standpoint. FunctionScoreQueryBuilder doesn't need to know of care that SearchModule is the thing that resolves the parser. If we felt like testing fromXContent without SearchModule we could implement a super easy mock implementation.

Finally, we don't need a whole class to wrap the HashMap - we can use a method. We could use a lambda if that was readable. It isn't in this context.

As to whether we're better off making our own functional interface or using one out of java I'm not sure. Using one strait from java saves some code and makes the argument more recognizable as a functional interface. We can (and I should have) documented it with Javadoc on the method. Using our own interface gives us a super obvious place to put the docs.